### PR TITLE
Add yaml outputs of resources to streaming logs and add import&update steps to uptest

### PR DESCRIPTION
**Pull request by sergenyalcin**
### Description of your changes

- This PR adds yaml outputs of resources to streaming log while waiting the Ready condition. Kuttl has a `script` field for supporting multiple commands.
- This PR adds new two test steps to uptest: import and update

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Tested locally.

### CODE REVIEW

File: internal/templates/01-update.yaml.tmpl
Comment (ulucinar) [inline internal/templates/01-update.yaml.tmpl:1]: We had better make this step conditioned on the `resource.UpdateField` not being empty. The rationale is for some resources we may not be able to find an appropriate parameter to update (e.g., any parameter we update might require the external resource to be replaced by Terraform). For such cases, we may just need to skip the resource upgrade tests.

File: internal/templates/01-update.yaml.tmpl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+commands:
+{{- range $resource := .Resources }}
+{{- if eq $resource.KindGroup "secret." -}}
+  {{continue}}
+{{- end -}}
+{{- if not $resource.Namespace }}
+- script: '${KUBECTL} patch {{ $resource.KindGroup }}/{{ $resource.Name }} --type=json -p "[{"op": "add", "path": "/spec/forProvider/{{ $resource.UpdateField }}", "value": {{ $resource.UpdateValue }}}]"'
```
Comment (ulucinar) [inline internal/templates/01-update.yaml.tmpl:9]: We can define a provider-level default for the parameter to be used in resource update tests. We could define this default in the reusable job definition as follows:
```yaml
jobs:
  open-bump-pr:
    uses: ...
    with:
        update-test-parameter: ...
```

File: internal/templates/01-update.yaml.tmpl
Comment (sergenyalcin) [reply internal/templates/01-update.yaml.tmpl:1]: Done!

File: internal/templates/01-update.yaml.tmpl (hunk 1 above)
Comment (sergenyalcin) [reply internal/templates/01-update.yaml.tmpl]: Done!

**Review by ulucinar — APPROVED** [review]
Thanks @sergenyalcin, lgtm. If we have a bug in the resource updates, let's not hide that as we have discussed.

File: internal/templates/01-update.yaml.tmpl [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+commands:
+{{- range $resource := .Resources }}
+{{- if eq $resource.UpdateParameter "" -}}
+  {{continue}}
+{{- end -}}
+{{- if eq $resource.KindGroup "secret." -}}
+  {{continue}}
+{{- end -}}
+{{- if not $resource.Namespace }}
+- command: ${KUBECTL} patch {{ $resource.KindGroup }}/{{ $resource.Name }} --type=merge -p '{"spec":{"forProvider":{{ $resource.UpdateParameter }}}}'
```
Comment (ulucinar) [inline internal/templates/01-update.yaml.tmpl:12]: We had better only patch the target manifest in the update tests.

**Pull request merged**